CHANGELOG.md

```diff
@@ -22,6 +22,8 @@ All notable changes to this project will be documented in this file.
 - BREAKING: Inject the vector aggregator address into the vector config using the env var `VECTOR_AGGREGATOR_ADDRESS` instead
     of having the operator write it to the vector config ([#704]).
 - test: Bump to Vector `0.46.1` ([#719]).
+- Previously this operator would hardcode the UID and GID of the Pods being created to 1000/0, this has changed now ([#718])
```

Comment:
I think we should mark this as potentially breaking.
I'm used to having Pod Security Policy in the k8s cluster that require a non-zero runAsUser for security reasons.
Nowadays there is Pod Security Admission, but I never used that.
Us removing the runAsUser would break existing installations on such clusters, so potentially breaking.

In general, playing a bit around with Pod Security Admission and check that we still work with them and documenting the outcome would be great.
But I don't know if we have the time for that

Reply:
I had the breaking in and removed it again because I couldn't think of a scenario where it would really be breaking

I _did_ look at the Pod Admission stuff and I am fairly certain that this PR doesn't affect us there.
The [restricted policy](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted) has a mention of runAsUser but it only says that it can't be 0 which will be the case.

So... I don't have a strong opinion. I can't think of anything this might break but if you want the label in there I'm also not against it. You decide :)

Reply:
I also don't have a strong opinion. I have a slight preference for better safe than sorry and marking it as breaking.
At the old clusters I used to work with this would have been a show-stopper, but the days of Pod Security Policy are gone.

Reply:
But good thing that you looked at Pod Admissions! :)

Reply:
Okay, happy to amend. But can you tell me what would be a showstopper exactly?
Nothing will change with this PR I believe.

Reply:
Pod Security Policies have been removed in 1.25, I don't think we should mention the and Pod Security Admission doesn't support what you're saying as far as I understand: https://kubernetes.io/docs/concepts/security/pod-security-standards/

I'm not against adding something but I believe your suggestion to be factually incorrect or incomplete.

Reply:
I intentionally wrote "policies" pretty vague :sweat_smile: But let's not over-complicate things and hope for the best. Ship it :)

Reply:
I do understand what you mean and I'll adapt the changelog. You're not wrong with your intention.

What about this?

```
- BREAKING: Previously this operator would hardcode the UID and GID of the Pods being created to 1000/0, this has changed now ([#718])
  - The `runAsUser` and `runAsGroup` fields will not be set anymore by the operator
  - The defaults from the docker images itself will now apply, which will be different from 1000/0 going forward
  - This is marked as breaking because tools and policies exist which require these fields to be set
```

Reply:
Sure, works for me. One minor thing

because tools and policies *might* exist